tests/aws/services/sns/test_sns.py

```diff
@@ -3810,7 +3811,14 @@ def _clean_headers(response_headers: dict):
             snapshot.match("http-message", payload)
             snapshot.match("http-message-headers", _clean_headers(notification_request.headers))
 
-        unsub_request = requests.get(expected_unsubscribe_url)
+        auth_header = aws_stack.mock_aws_request_headers(
+            "sns", region_name=TEST_AWS_REGION_NAME, access_key=TEST_AWS_ACCESS_KEY_ID
+        )["Authorization"]
+
+        unsub_request = requests.get(
+            url=expected_unsubscribe_url, headers={"Authorization": auth_header}
+        )
```

Comment:
I think this is wrong, and that the issue is in SNS `Unsubscribe`, because when working/validating the test in AWS this will fail because the credentials are wrong.

It seems `Unsubscribe` is an operation that can be done unauthenticated.
https://docs.aws.amazon.com/sns/latest/api/API_Unsubscribe.html

However, we have an issue here, is that requests always fall back to some credentials, so we won't be able to fully implement the behaviour for now that the documentation is explaining (namely the fact that SQS unsubscribe should always be authenticated, because for us, they always are so we can't reject an unauthenticated one).

But we can fix Unsubscribe, as it works the same way as `ConfirmSubscription`: 
```python
        # this request can come from any http client, it might not be signed (we would need to implement
        # `authenticate_on_unsubscribe` to force a signing client to do this request.
        # so, the region and account_id might not be in the request. Use the ones from the topic_arn
        try:
            parsed_arn = parse_arn(topic_arn)
        except InvalidArnException:
            raise InvalidParameterException("Invalid parameter: Topic")

        store = self.get_store(account_id=parsed_arn["account"], region_name=parsed_arn["region"])
```
This above should be also be in `Unsubscribe`, to correctly get the account/region for the store to then actually do the work.

Reply:
Thanks for pointing that out @bentsku.